boot/boottest/bootenv.go

```diff
@@ -110,12 +120,21 @@ func (b16 Bootenv16) SetRollbackAcrossReboot(which []snap.Type) error {
 	return nil
 }
 
-// RunBootenv16 implements manipulating a UC20 run-mode boot env for
+// RunBootenv20 implements manipulating a UC20 run-mode boot env for
 // testing.
 type RunBootenv20 struct {
 	*bootloadertest.MockExtractedRunKernelImageBootloader
 }
 
+// MockUC20EnvRefExtractedKernelRunBootenv wraps a mock bootloader for UC20 run-mode boot
+// env manipulation.
+func MockUC20EnvRefExtractedKernelRunBootenv(b *bootloadertest.MockBootloader) *Bootenv16 {
```

Comment:
we want a TODO about avoiding Bootenv16 at some point

Reply:
is this TODO about just not returning a Bootenv16, or about not using Bootenv16 at all in this case?

Reply:
as discussed on IRC, this was about not returning Bootenv16, and this is done now